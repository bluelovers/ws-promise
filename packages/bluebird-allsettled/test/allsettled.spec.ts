import Bluebird from 'bluebird';
import Bluebird36 from 'bluebird36';
import { allSettled } from '../index';

test(`allSettled`, async () =>
{

	let actual = await allSettled([
		helloTimeout('juan', 2000),
		helloTimeout('valentina', 1000, true),
		helloTimeout('julian', 3000),
	]);

	expect(actual).toStrictEqual([
		{ status: 'fulfilled', value: 'juan' },
		{ status: 'rejected', reason: 'valentina' },
		{ status: 'fulfilled', value: 'julian' }
	]);
	expect(actual).toMatchSnapshot();

	function helloTimeout<N>(name: N, timeout: any, bool?: any)
	{
		if (bool)
		{
			return Bluebird.reject(name)
		}
		else
		{
			return Bluebird.delay(timeout)
				.then(() =>
				{
					return Bluebird.resolve(name)
				})
		}
	}


});

test(`allSettled:Bluebird36`, async () =>
{

	let actual = await allSettled([
		helloTimeout('juan', 2000),
		helloTimeout('valentina', 1000, true),
		helloTimeout('julian', 3000),
	], {
		promise: Bluebird36,
	});

	expect(actual).toStrictEqual([
		{ status: 'fulfilled', value: 'juan' },
		{ status: 'rejected', reason: 'valentina' },
		{ status: 'fulfilled', value: 'julian' }
	]);
	expect(actual).toMatchSnapshot();

	function helloTimeout<N>(name: N, timeout: any, bool?: any)
	{
		if (bool)
		{
			return Bluebird36.reject(name)
		}
		else
		{
			return Bluebird36.delay(timeout)
				.then(() =>
				{
					return Bluebird36.resolve(name)
				})
		}
	}

});
