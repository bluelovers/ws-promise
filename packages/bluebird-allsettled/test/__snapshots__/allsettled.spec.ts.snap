// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allSettled 1`] = `
Array [
  Object {
    "status": "fulfilled",
    "value": "juan",
  },
  Object {
    "reason": "valentina",
    "status": "rejected",
  },
  Object {
    "status": "fulfilled",
    "value": "julian",
  },
]
`;

exports[`allSettled:Bluebird36 1`] = `
Array [
  Object {
    "status": "fulfilled",
    "value": "juan",
  },
  Object {
    "reason": "valentina",
    "status": "rejected",
  },
  Object {
    "status": "fulfilled",
    "value": "julian",
  },
]
`;
